### PR TITLE
feat(github-ci): add GitHub Action to install dfx

### PR DESCRIPTION
**Pull request by mmicu**
# Description

Add a GitHub Action that installs `dfx` in GitHub Action Workflow.

# How Has This Been Tested?

`act` can be used to test this new action, together with `.github/workflows/tests/dfx-action.yml`:

```
$ act -W .github/workflows/tests/dfx-action.yml
```

### CODE REVIEW

File: .github/workflows/tests/dfx-action.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+on:
+  push:
+    branches:
+      - 'main'
+    paths:
+      - '.github/workflows/tests/dfx-action.yml'
+      - 'action.yml'
+  pull_request:
+    paths:
+      - '.github/workflows/tests/dfx-action.yml'
```
Comment (ericswanson-dfinity) [inline .github/workflows/tests/dfx-action.yml:10]: I notice this didn't run for this PR.  Does the file need to be in the workflows directory, like `.github/workflows/dfx-action.yml` ?

Comment (mmicu) [reply .github/workflows/tests/dfx-action.yml]: I moved the file, but I cannot see any run as well. Do we need any additional GitHub configuration?

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  using: composite
+  steps:
+    - name: Install dfx
+      shell: sh
+      run: |
+        echo "Version ${{ inputs.dfx-version }} will be installed"
+        if [ "${{ inputs.dfx-version }}" = "latest" ]; then
+          sh -ci "$(curl -fsSL https://internetcomputer.org/install.sh)"
+        else
+          DFX_VERSION=${{ inputs.dfx-version }} sh -ci "$(curl -fsSL https://internetcomputer.org/install.sh)"
```
Comment (ericswanson-dfinity) [inline action.yml:21]:
```suggestion
          DFX_VERSION="${{ inputs.dfx-version }}" sh -ci "$(curl -fsSL https://internetcomputer.org/install.sh)"
```

File: .github/workflows/dfx-action.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+on:
+  push:
+    branches:
+      - 'main'
+    paths:
+      - '.github/workflows/dfx-action.yml'
+      - 'action.yml'
```
Comment (ericswanson-dfinity) [inline .github/workflows/dfx-action.yml:7]: We should set this up as a required check.

There are two options for this:
- run on all PRs (remove these path qualifiers)
- add a second workflow with the same name that passes if this one doesn't run (paths-ignore)

For a single-focus repo like this one, I'm not sure it's worth it to try to limit when this workflow runs, so I'd recommend to remove the path qualifiers.

```suggestion
```

File: .github/workflows/dfx-action.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  pull_request:
+    paths:
+      - '.github/workflows/dfx-action.yml'
+      - 'action.yml'
```
Comment (ericswanson-dfinity) [inline .github/workflows/dfx-action.yml:11]:
```suggestion
```

**Review by ericswanson-dfinity — CHANGES_REQUESTED** [review]

**Review by ericswanson-dfinity — APPROVED** [review]

File: .github/workflows/tests/dfx-action.yml (hunk 1 above)
Comment (ericswanson-dfinity) [reply .github/workflows/tests/dfx-action.yml]: Workflow ran and passed after merge.  Thanks @mmicu!

**Pull request merged**